src/index.d.ts

```diff
@@ -69,7 +69,7 @@ type ModuleAPI<Registry, PublicAPI extends Array<keyof Registry> = []> = {
 type ProviderFnArgs<Registry extends ObjectLike> = {
   [key in keyof ExternalDeps<Registry>]:
     | ExternalDeps<Registry>[key]
-    | ((arg?: FlatDependencyTree<Registry>) => ExternalDeps<Registry>[key]);
+    | ((arg?: InjectableMap<Registry>) => ExternalDeps<Registry>[key]); // technically there is no constraint on arg, it is quite the opposite: it may bring new constraints but this gets contrived.
```

Comment:
I think it should: 

```Javascript
const createModule = createProvider({
    injectables:{
        someThing({random}: {random:number}) { return random * 42} 
    },
})

const myModule = createModule({
    random() {
        return Math.random();
    }
})
```

should be ok

Reply:
Yes, if you make it mandatory your example above is still valid as 

```typescript
() => number
```

is compatible with

```typescript
(arg: InjectableMap<Registry>) => ExternalDeps<Registry>['random']
```

On the other hand, this is not possible:

```typescript
const createModule = createProvider({
    injectables:{
        someThing: ({ random }: { random: number }) => random * 42
        randomMin: () => 100
    },
})

const myModule = createModule({
    random(injectables) => injectables?.randomMin +Math.random()
})
```

above, the `injectables` type is `InjectableMap<Registry> | undefined`, so we have to use the `?.` operator to access the value even if we know it will be defined (also because of that we can't destructure it).